.github/workflows/near_crates_publish.yml

```diff
@@ -11,20 +13,29 @@ on:
 jobs:
   publish-cargo-crates:
     name: "Publish near-workspaces on crates.io https://crates.io/crates/near-workspaces"
-    runs-on: "ubuntu-22.04-16core"
+    runs-on: "ubuntu-20.04-16core"
     environment: deploy
     permissions:
       contents: write # required for crates push
     timeout-minutes: 30 
     
     steps:
       - name: Checkout near/nearcore's ${{ github.event.inputs.branch }} branch
+        if: ${{ github.event_name == 'workflow_dispatch'}}
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
           ref: ${{ github.event.inputs.branch }}
+
+      - name: Checkout nearcore repository
+        if: ${{ github.event_name != 'workflow_dispatch'}}
+        uses: actions/checkout@v4
```

Comment:
@andrei-near, can you confirm that this will pickup the right branch/tag when it is triggered by the release?

Reply:
The docs say that `ref` is set to the branch that it was called from, in this case, the release branch.
https://github.com/actions/checkout/blob/main/action.yml#L8-L11